src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingServer.kt

```diff
@@ -21,22 +21,28 @@ import java.util.Stack
 
 
 class OpenTracingServer(
-        val filters: List<(ApplicationCall) -> Boolean>
+        val filters: List<(ApplicationCall) -> Boolean>,
+        val tagsToExtractFromPath: Map<String, Regex>
 ) {
     class Configuration {
         val filters = mutableListOf<(ApplicationCall) -> Boolean>()
+        val tagsToExtractFromPath: MutableMap<String, Regex> = mutableMapOf(uuidTagAndReplace)
 
         fun filter(predicate: (ApplicationCall) -> Boolean) {
             filters.add(predicate)
         }
+
+        fun extractTagFromPath(tagName: String, regex: Regex) {
```

Comment:
I prefer 'replace' over extract, which is less common. How about 'replaceInPathAndTagSpan' to be really explicit (this is complex config).

Reply:
I started with replace and @hj1611 didn't like it 😂